src/hb-ot-layout.cc

```diff
@@ -1009,43 +1027,109 @@ hb_ot_layout_get_size_params (hb_face_t    *face,
 			      unsigned int *range_start,       /* OUT.  May be nullptr */
 			      unsigned int *range_end          /* OUT.  May be nullptr */)
 {
```

Comment:
Extracted _get_gsubgpos_matched_feature_params from hb_ot_layout_get_size_params. The difference is however it searches gsub table also which is negligible I guess but easily can be fixed.